certbot/plugins/webroot.py

```diff
@@ -217,16 +219,14 @@ def cleanup(self, achalls):  # pylint: disable=missing-docstring
                 os.remove(validation_path)
                 self.performed[root_path].remove(achall)
 
-        for root_path, achalls in six.iteritems(self.performed):
-            if not achalls:
```

Comment:
I think we actually still need a check like this `if` statement here as we should only be cleaning up the challenges given in the `achalls` parameter. In practice, I don't think this matters right now, but the [interface](https://certbot.eff.org/docs/api/interfaces.html#certbot.interfaces.IAuthenticator.cleanup) only guarantees `achalls` is a non-empty subset of previously performed challenges. We could make use of this in the future so we should make sure there are no active challenges in a directory before trying to delete it.

Reply:
fixed this and added a test for it, though it's done implicitly because `rmdir` doesn't execute if the directory is not empty.